src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -1494,15 +1507,18 @@ public boolean canSee(Player player)
 	@Override
 	public boolean isFlying()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return flying;
 	}
 
 	@Override
 	public void setFlying(boolean value)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		PlayerToggleFlightEvent event = new PlayerToggleFlightEvent(this, value);
+		Bukkit.getPluginManager().callEvent(event);
+		if (!event.isCancelled())
+		{
+			this.flying = event.isFlying();
+		}
```

Comment:
This is an erroneous implementation.
CraftBukkit does not call such an event when invoking this method.
Setters only modify the state whereas any PlayerEvent is the result of the actions from an actual Player, you would need a simulation method for this. The setter itself doesn't call an event.